Sources/Misc/Concurrency/SynchronizedLargeItemCache.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SynchronizedLargeItemCache.swift
+//
+//  Created by Jacob Zivan Rakidzich on 10/9/25.
+
+import Foundation
+
+/// A thread-safe wrapper around `LargeItemCacheType` for synchronized file-based caching operations.
+internal final class SynchronizedLargeItemCache {
+
+    private let cache: LargeItemCacheType
+    private let lock: Lock
+    private let documentURL: URL?
+
+    init(cache: LargeItemCacheType, basePath: String) {
+        self.cache = cache
+        self.lock = Lock(.nonRecursive)
+        self.documentURL = cache.createDocumentDirectoryIfNeeded(basePath: basePath)
+    }
+
+    /// Performs a synchronized read operation
+    internal func read<T>(_ action: (LargeItemCacheType, URL?) throws -> T) rethrows -> T {
```

Comment:
Sorry, I said `internal` in my previous comment when I meant `private` (I meant "internal for the `SynchronizedLargeItemCache` 😅 )
```suggestion
    private func read<T>(_ action: (LargeItemCacheType, URL?) throws -> T) rethrows -> T {
```